src/box/lua/trigger.c

```diff
@@ -74,20 +135,62 @@ luaT_trigger_call(struct lua_State *L)
 	struct event *event = event_get(event_name, false);
 	if (event == NULL)
 		return 0;
-	int top = lua_gettop(L);
-	int narg = top - 1;
+	int nargs = lua_gettop(L) - 1;
+
+	/*
+	 * If the function is called on Lua-stack from fiber storage, create
+	 * new thread because we cannot dump port_lua to the same stack as
```

Comment:
Can you make `port_lua` work with the same stack?

Reply:
Decided to move this check to `func_adapter_lua_call`: `func_adapter_lua_call` will check if the port storing arguments is a Lua port and if it is and points to the current fiber Lua stack, create a new stack for it.

Reply:
Done.
By the way, now one empty on_replace trigger slows Tarantool down by 27%, but without the optimization slowdown is about 42%.